### PR TITLE
#217 remove total after tax from mutations

### LINKED ISSUE #217: Remove total_after_tax from mutations
Linked to #214

### PR DESCRIPTION
**Pull request by roxy-dao**
Closes #217 

Remove total after tax from mutations and reflected changes in F/E

### CODE REVIEW

File: server/service/src/invoice_line/outbound_shipment_line/insert/generate.rs [hunk 1]
```diff
@@ -79,7 +78,11 @@ fn generate_line(
         item_code,
         stock_line_id: Some(stock_line_id),
         total_before_tax,
-        total_after_tax,
+        total_after_tax: if tax.is_some() {
+            total_before_tax * (f64::from(1) + tax.unwrap() / 100.0)
+        } else {
+            total_before_tax
+        },
```
Comment (clemens-msupply) [inline server/service/src/invoice_line/outbound_shipment_line/insert/generate.rs:85]: You can avoid unwrap by doing:

```rust 
total_after_tax: match tax {
  Some(tax) => total_before_tax * (1.0 + tax / 100.0)
  None => total_before_tax
},
```
Also note how you can avoid f64::from

Worth having a helper for It? e.g. `fn totalAfterTax(total_before_tax: f64, tax: Option<f64>) -> f64`

File: server/service/src/invoice_line/outbound_shipment_service_line/update/generate.rs [hunk 2]
```diff
@@ -58,6 +53,9 @@ pub fn generate(
         update_line.note = Some(note);
     }
 
+    update_line.total_after_tax =
+        update_line.total_before_tax * (f64::from(1) + update_line.tax.unwrap_or_default() / 100.0);
```
Comment (clemens-msupply) [inline server/service/src/invoice_line/outbound_shipment_service_line/update/generate.rs:57]: Use the same way to calculate total_before_tax (see other comment)

File: server/service/src/invoice_line/inbound_shipment_line/update/generate.rs [hunk 3]
```diff
@@ -90,7 +90,7 @@ fn generate_line(
     }
 
     update_line.total_after_tax =
-        update_line.cost_price_per_pack * update_line.number_of_packs as f64;
+        update_line.total_before_tax * (f64::from(1) + update_line.tax.unwrap_or_default() / 100.0);
```
Comment (clemens-msupply) [inline server/service/src/invoice_line/inbound_shipment_line/update/generate.rs:93]: Thats what we discussed last time, right?

**Review by clemens-msupply — CHANGES_REQUESTED** [review]
Looks good could you please remove the unwraps() (see comment) and use the same way to calculate totalAfterTax once you use is_some() and then you use unwrap_or_default.

File: server/service/src/invoice_line/inbound_shipment_line/update/generate.rs (hunk 3 above)
Comment (roxy-dao) [reply server/service/src/invoice_line/inbound_shipment_line/update/generate.rs:93]: Yup~

File: server/service/src/invoice_line/inbound_shipment_line/insert/generate.rs [hunk 4]
```diff
@@ -67,10 +67,9 @@ fn generate_line(
         item_code,
         stock_line_id: None,
         total_before_tax,
-        total_after_tax: if tax.is_some() {
-            total_before_tax * (f64::from(1) + tax.unwrap() / 100.0)
-        } else {
-            total_before_tax
+        total_after_tax: match tax {
```
Comment (andreievg) [inline server/service/src/invoice_line/inbound_shipment_line/insert/generate.rs:70]: another way (not saying you should change) is more a functional style. 
```
 tax.map(|t| total_before_tax * (1.0 + tax / 100.0))
            .unwrap_or(total_before_tax)
```

File: server/service/src/invoice_line/inbound_shipment_service_line/update/generate.rs [hunk 5]
```diff
@@ -31,7 +32,7 @@ pub fn generate(
         None
     };
 
-    let mut update_line = existing_line;
+    let mut update_line = existing_line.clone();
```
Comment (clemens-msupply) [inline server/service/src/invoice_line/inbound_shipment_service_line/update/generate.rs:35]: nit: Why do you need to clone it now? can it be avoided?

**Review by clemens-msupply — APPROVED** [review]
thanks for the changes

File: server/service/src/invoice_line/inbound_shipment_service_line/update/generate.rs (hunk 5 above)
Comment (roxy-dao) [reply server/service/src/invoice_line/inbound_shipment_service_line/update/generate.rs:35]: Before I had some ownership problem but can add it back now with the changes

**Pull request merged**